ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -412,72 +414,96 @@ def test011_analysis_number_filter_and_export(self):
             'Check if export of order has analyis number = {}  '.format(analysis_number))
         self.assertIn(analysis_number, sheet_values)
 
-    # will change that the duplicate many copies will be from the the child table not from the active table     
     def test012_duplicate_many_orders(self):
         """
-        New: Orders: Duplication from active table Approach: When I duplicate order 5 times, it will create 5 analysis records with the same order number
+        Make sure that the user can duplicate suborder with multiple copies ( record with test units )
+        Make sure that the user can duplicate suborder with multiple copies ( record with test plans )
         LIMS-4285
+        LIMS-6224
         :return:
         """
-        number_of_copies = randint(2, 5)
-        self.base_selenium.LOGGER.info(' Select Random Order')
-        selected_row = self.order_page.get_random_order_row()
-        selected_order_data = self.base_selenium.get_row_cells_dict_related_to_header(
-            row=selected_row)
-        self.order_page.click_check_box(source=selected_row)
+        no_of_copies = randint(2, 5)
+        self.base_selenium.LOGGER.info('get last order row that you created')
+        all_orders= self.orders_api.get_all_orders().json()['orders']
+        record_id = randint(0, len(all_orders)-2)
+        data_before_duplicate_main_order= all_orders[record_id]
+
+        order_no = data_before_duplicate_main_order['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no,
+                                              field_type='text')
+
+        self.order_page.get_child_table_data(index=0)
+        rows_with_childtable = self.order_page.result_table(element='general:table_child')
+        for row in rows_with_childtable[:-1]:
+            data_before_duplicate_sub_order = self.base_selenium.get_row_cells_dict_related_to_header(row=row,
```

Comment:
for each iteration, the value of `data_before_duplicate_sub_order` gonna be overwritten

Reply:
@0xIslamTaha yes

Reply:
@dianaeldemerdash so why do u do it?

Reply:
@0xIslamTaha I got it, we have missing point, data before duplicate didn't overwritten, it overwritten after the duplicate not before so this line the value didn't overwritten

Reply:
@dianaeldemerdash 
u r doing something like this 
```python
for x in range(5):
     y=x
```
so at the end, y will be equal to 4 in all cases > so what is the point of the loop?

Reply:
@0xIslamTaha yes, so we will reach to the same results in all cases